src/ray/raylet/worker_pool.cc

```diff
@@ -556,6 +579,10 @@ void WorkerPool::OnWorkerStarted(const std::shared_ptr<WorkerInterface> &worker)
     auto &io_worker_state = GetIOWorkerStateFromWorkerType(worker_type, state);
     io_worker_state.registered_io_workers.insert(worker);
     io_worker_state.num_starting_io_workers--;
+  } else {
+    // All task/actor workers should have a runtime env hash computed from
+    // their job ID and (optionally) custom runtime environment.
+    RAY_CHECK(worker->GetRuntimeEnvHash());
```

Comment:
Hmm so the case where we return "0" for runtime hash can never get to here?

Reply:
Yeah, it would only happen if a worker process that wasn't started by the raylet somehow registered. I'll add an error message to the check.